py/aou_workbench_client/cohorts.py

```diff
@@ -23,18 +23,14 @@ def materialize_cohort(request, max_results=None):
     
     Multiple server requests may be made to retrieve all the results,
     using the page_size specified in the request for each request."""
-    client = get_authenticated_swagger_client()
-    cohorts_api = CohortsApi(api_client=client)
     num_results = 0
     # Clone the request, since we're going to be modifying it.
     request = deepcopy(request)    
     request.page_size = request.page_size or 1000
     while True:
       if max_results and (max_results - num_results) < request.page_size:
         request.page_size = max_results - num_results   
-      response = cohorts_api.materialize_cohort(all_of_us_config.workspace_namespace,
-                                                all_of_us_config.workspace_id,
-                                                request=request)      
+      response = materialize_cohort_page(request)      
```

Comment:
This looks OK, but I'm slightly nervous that researchers using get_authenticated_swagger_client() directly will run into the same problem. (This assumes that in future we will support get_authenticated_swagger_client as an API researchers could call in order to use our swagger client directly, rather than having to wrap all our API calls in helper methods.)

So a thought: what if we extend ApiClient with a class that checks before each call to see if the access token being used by the client has expired, and if so refreshes the token then. Then we can return an instance of that class instead of ApiClient from get_authenticated_swagger_client(), and it should take care of keeping the auth token fresh no matter how it's used.

WDYT? (If you like the idea but don't have time, I'm happy to take a stab.)

Reply:
Or we could file that as a TODO; I'm pretty sure no researcher is going to use get_authenticated_swagger_client() right now.

Reply:
Definitely agree with proposed approach. Added a TODO for now; want to make sure this make it into the next release and would want to test the other thing more thoroughly.